packages/assets-controllers/src/NftController.ts

```diff
@@ -512,6 +512,53 @@ export class NftController extends BaseController<
     });
   }
 
+  /**
+   * Fetches NFT Collection Metadata from the NFT API.
+   *
+   * @param contractAddresses - The contract addresses of the NFTs.
+   * @param chainId - The chain ID of the network where the NFT is located.
+   * @returns NFT collections metadata.
+   */
+  async fetchNftCollectionsMetadata(
+    contractAddresses: string[],
+    chainId: Hex,
```

Comment:
Should we future-proof and support different contracts on different chains?

We'd need to accept an array of requests for example, assuming this is even supported by the API.

Reply:
It does accept array of chains for `/tokens` but `/collections` are bound to one `chainId`
https://nft.api.cx.metamask.io/docs#/Account-V1/NftV1Controller_getCollections
Did I miss something here?

Reply:
That is true, the /collections API supports a single chainId for now,
Would you need it to support multiple chainIds for your usecase? 
Or maybe using the /tokens api instead like suggested [here](https://github.com/MetaMask/core/pull/4506#discussion_r1669934584) would work better for you?

Reply:
I can't say we will never need it but as far as I know, cross chain simulations is not possible today. So `/collections` path will be enough for us now.